pipelines/productionBuild.yml

```
@@ -1,220 +1,272 @@
-                  "operationSetCode": "NuGetSign",
-                  "parameters": [ ],
-                  "toolName": "sign",
-                  "toolVersion": "1.0"
-              },
-              {
-                  "keyCode": "CP-401405",
-                  "operationSetCode": "NuGetVerify",
-                  "parameters": [ ],
-                  "toolName": "sign",
-                  "toolVersion": "1.0"
-              }
-          ]
-    SessionTimeout: 20
-
-- task: CopyFiles@2
-  displayName: 'Copy release scripts to artifact staging directory'
-  inputs:
-    SourceFolder: '$(Build.SourcesDirectory)'
-    Contents: |
-     scripts\GetNugetPackageVersion.ps1
-     scripts\GetLatestCommitSHA.ps1
-    TargetFolder: '$(Build.ArtifactStagingDirectory) '
-
-- task: PublishBuildArtifacts@1
-  displayName: 'Publish Artifact: Microsoft.Graph.Core.nupkg and release pipeline scripts'
-  inputs:
-    PathtoPublish: '$(Build.ArtifactStagingDirectory)'
-    ArtifactName: ProductionReleaseArtifact
-
+    - stage: deploy
+      condition: and(contains(variables['build.sourceBranch'], 'refs/heads/master'), succeeded())
+      dependsOn: build
+      jobs:
+        - deployment: deploy_nuget
+          pool:
+            name: Azure-Pipelines-1ESPT-ExDShared
+            os: windows
+            image: windows-latest
+          dependsOn: []
+          environment: nuget-org
+          strategy:
+            runOnce:
+              deploy:
+                steps:
+                - task: NuGetToolInstaller@1
+                  displayName: 'Use NuGet >=5.2.0'
+                  inputs:
+                    versionSpec: '>=5.2.0'
+                    checkLatest: true
+                - task: DownloadPipelineArtifact@2
+                  displayName: Download nupkg from artifacts
+                  inputs:
+                    artifact: ProductionReleaseArtifact
+                    source: current
+                - task: PowerShell@2
+                  displayName: 'Get Latest Commit SHA from repo'
+                  inputs:
+                    targetType: filePath
+                    filePath: '$(Pipeline.Workspace)\scripts\GetLatestCommitSHA.ps1'
+                    arguments: '-repo "msgraph-sdk-dotnet-core" -owner "microsoftgraph" -branchName "master"'
+                    pwsh: true
+                - task: PowerShell@2
+                  displayName: 'Extract release information to pipeline'
+                  inputs:
+                    targetType: 'filePath'
+                    filePath: $(Pipeline.Workspace)\scripts\GetNugetPackageVersion.ps1
+                    pwsh: true
+                    arguments: '-packageDirPath "$(Pipeline.Workspace)/"'
+                - task: 1ES.PublishNuget@1
+                  displayName: 'Push release to NuGet.org'
+                  inputs:
+                    command: push
+                    packagesToPush: '$(Pipeline.Workspace)\Microsoft.Graph.Core.*.nupkg'
```

```suggestion
                    packageParentPath: '$(Pipeline.Workspace)'
                    packagesToPush: '$(Pipeline.Workspace)\Microsoft.Graph.Core.*.nupkg'
```

I did have to add this for kiota so the deploy would work